src/zjs_util.h

```diff
@@ -107,6 +107,30 @@ uint32_t zjs_uncompress_16_to_32(uint16_t num);
 
 void zjs_print_error_message(jerry_value_t error);
 
+/**
+ * Release a jerry_value_t passed by reference
+ */
+void zjs_free_value(jerry_value_t *value);
+
+/**
+ * Macro to declare a jerry_value_t and have it released automatically
+ *
+ * If you're going to be returning the value from a function, do not use this
+ * as ownership will transfer to the caller (unless you acquire it on return).
+ * Also don't use this for a variable that holds a C copy of a value that was
+ * passed into you, as those are owned by the caller.
+ *
+ * NOTE: If you reuse the variable in your function, you will still need to
+ *   release the old value before you lose it.
+ *
+ * NOTE: If you want to return a value created with ZVAL, you need to call
+ *   jerry_acquire_value() on it first, typically in the return line.
+ *
+ * NOTE: Be careful to always initialize this value, or else an uninitialized
+ *   value will get freed. (Or will compiler catch that?)
+ */
+#define ZVAL(name) jerry_value_t name __attribute__ ((__cleanup__(zjs_free_value)))
```

Comment:
Personally, I'd much rather be able to write:

```
ZVAL xyz = jerry_create_object();
```

So the `#define` would become:

```
#define ZVAL const jerry_value_t __attribute__ ((__cleanup__(zjs_free_value)))
```

I changed it to `const`, to prevent you from accidentally doing:

```
ZVAL xyz = jerry_create_object();
xyz = jerry_create_object(); // whoops, leaking!
```

Lastly, you probably also want a macro for "unused" `jerry_value_t` return values, so something along these lines:

```
#define CONCAT_INNER(x, y) x##y
#define CONCAT(x, y) CONCAT_INNER(x, y)
#define ZVAL_UNUSED ZVAL CONCAT(zval_unused, __COUNTER__)

ZVAL_UNUSED = jerry_call_function(...); // say I don't care about the return value
```